### ISSUE TITLE
[JS-only] Support <react-native-snap-carousel>

### ISSUE THREAD

**Issue by robertev** [microsoft/react-native-windows#4153]
# Support react-native-snap-carousel on RNW vNext

## Basic Info
- Module Repo URL: https://github.com/archriss/react-native-snap-carousel
- Target version: ^3.8.4
- [ ] Issue exists on module repo (if so, [link]())
## Overview of Required Functionality
Swiper/carousel   component for React Native featuring previews, multiple layouts, parallax   images, performant handling of huge numbers of items, and more. Compatible   with Android & iOS.
archriss
PureJS - yes

**Comment by harinikmsft** [comment]
From hacking on this module during our recent hackathon, here are my observations:

- It is definitely a JS only module and did not need any extra plumbing to get it to work on react-native-windows. I was able to drop it in and "something showed up"
- There are some layout issues that I was unable to resolve with any kind of styling/layout changes. My guess is the module's layout engine is fighting with yoga+XAML's layout engine causing items, cards to overlap.
- The animations seemed to be working. I tried setting `autoplay = {true}` to just see if there was some kind of sliding motion and there was - items moved with animations horizontally. I was not able to get the `vertical={true}` mode to work at all. 
- The biggest problem is there are no non-touch modes for this control, so - we might have to add mouse, keyboard support in the form of physical buttons or scroller of some kind to show up etc., This will need some PM design thinking.
